core/src/main/java/io/questdb/std/str/Utf8s.java

```diff
@@ -1203,48 +1212,58 @@ private static int encodeUtf16Surrogate(@NotNull Utf8Sink sink, char c, @NotNull
         return pos;
     }
 
-    // Left hand has to be lower-case.
-    private static boolean equalsAsciiLowerCase(@NotNull Utf8Sequence lLC, @NotNull Utf8Sequence r, int size) {
-        for (int i = 0; i < size; i++) {
-            if (lLC.byteAt(i) != toLowerCaseAscii(r.byteAt(i))) {
+    private static boolean equalPrefixBytes(
+            @NotNull Utf8Sequence l, long lSixPrefix, @NotNull Utf8Sequence r, long rSixPrefix, int prefixSize
+    ) {
+        long prefixMask = (1L << 8 * Math.min(VARCHAR_INLINED_PREFIX_BYTES, prefixSize)) - 1;
+        if (((lSixPrefix ^ rSixPrefix) & prefixMask) != 0) {
+            return false;
+        }
+        int i = VARCHAR_INLINED_PREFIX_BYTES;
+        for (; i <= prefixSize - Long.BYTES; i += Long.BYTES) {
+            if (l.longAt(i) != r.longAt(i)) {
+                return false;
+            }
+        }
+        for (; i < prefixSize; i++) {
+            if (l.byteAt(i) != r.byteAt(i)) {
                 return false;
             }
         }
         return true;
     }
 
-    // Left hand has to be lower-case.
-    private static boolean equalsAsciiLowerCase(@NotNull Utf8Sequence lLC, @NotNull Utf8Sequence r, int rLo, int rHi) {
-        int ls = lLC.size();
-        if (ls != rHi - rLo) {
-            return false;
-        }
-
-        for (int i = 0; i < ls; i++) {
-            if (lLC.byteAt(i) != toLowerCaseAscii(r.byteAt(i + rLo))) {
+    private static boolean equalSuffixBytes(
+            @NotNull Utf8Sequence seq, @NotNull Utf8Sequence suffix, int seqSize, int suffixSize
+    ) {
+        int seqLo = seqSize - suffixSize;
+        for (int i = 0; i < suffixSize; i++) {
```

Comment:
nit: we could do `longAt` main loop + `byteAt` tail here too like it's done in `equalPrefixBytes`.